gh-92936: update `http.cookies` docs post GH-113663 (#137566)
python/cpython: Doc/whatsnew/3.15.rst

  (Contributed by Alexander Enrique Urieles Nieto in :gh:`131724`.)

http.cookies
------------

> * Allow `"` (double quotes) in cookie values with a more linient regex for

Reviewer: ```suggestion
* Allow ``"`` (double quotes) in cookie values with a more linient regex for
```

Author: ah thanks!

>   doublequoted strings.

Reviewer: ```suggestion
  double-quoted strings.
```

> * Allow ``"`` double quotes in cookie values with a more linient regex for
>   double-quoted strings.

Reviewer 2: Can we be more specific about what's changed than "a more lenient regex"? For example, do we now properly conform to an RFC/web specification? Are there restrictions that still exist on double-quoted strings?
```suggestion
* Allow '``"``' double quotes in cookie values with a more lenient regex for
  double-quoted strings.
```

> * Allow '``"``' double quotes in cookie values with a more lenient regex for
>   double-quoted strings.

Author: ```suggestion
* Allow '``"``' double quotes in cookie values with a more lenient regex for
  double-quoted strings by not explicitly excluding a quote inside the value part of a cookie.
```

It's pretty similar to what was there before, just doesn't exclude the quotes: https://github.com/python/cpython/pull/113663/files#diff-e49aab1421911b035542d1a9221c31b8c74053f0e8b28f90f03af66177026176R429

Reviewer 3: This is becoming little too detailed for a whatsnew item. Since it is both a bugfix and behavior change (ref https://github.com/python/cpython/pull/137566#discussion_r2264066864) 

We could say - "Allow '``"``' double quotes in cookie values" or the existing detail is fine with me. 

@AA-Turner, what is your suggestion here?

Author: Personally I'm a fan of the shorter 
> "Allow '"' double quotes in cookie values"

Reviewer 3: I am good with the shorter version too. It is clear without confusing on the internal details like lenient regex (which the reader of whatsnew doc doesn't need to bother about) and if they really want they could refer to git log, bug report and other discussions.

>   (Contributed by Nick Burns and Senthil Kumaran in :gh:`137566`.)

Reviewer: ```suggestion
  (Contributed by Nick Burns and Senthil Kumaran in :gh:`92936`.)
```
